src/components/calcite-rating/calcite-rating.tsx

```diff
@@ -0,0 +1,356 @@
+          if (isLastItem) this.focusFirstItem();
+          else this.focusNextItem(itemToFocus);
+          break;
+        case "ArrowLeft":
+          if (isFirstItem) this.focusLastItem();
+          else this.focusPrevItem(itemToFocus);
+          break;
+        case "Home":
+          this.focusFirstItem();
+          e.preventDefault();
+          break;
+        case "End":
+          this.focusLastItem();
+          e.preventDefault();
+          break;
+        case "Tab":
+          if (isLastItem || (isFirstItem && e.shiftKey)) this.resetHoverState();
+          break;
+      }
+    }
+  }
+
+  private determineRatingPosition(e) {
+    if (!this.readOnly) {
+      this.value = e.target.dataset.value;
+      this.emitRatingChange();
+      this.determineActiveItems();
+    }
+  }
+
+  private determineInitialRating() {
+    const valueToUse = this.value ? this.value : this.average;
+    this.ratingItems?.forEach((item) => {
+      item.dataset.hovered = "false";
+      item.dataset.average =
+        this.average && !this.value && parseInt(item.dataset.value) <= this.average
+          ? "true"
+          : "false";
+      item.dataset.selected =
+        item.dataset.average === "false" && parseInt(item.dataset.value) <= valueToUse
+          ? "true"
+          : "false";
+      item.icon =
+        parseInt(item.dataset.value) <= valueToUse || this.readOnly
+          ? this.selectedIconType
+          : this.iconType;
+    });
+
+    if (this.average) {
+      this.determinePartialStar();
+    }
+  }
+
+  private determinePartialStar() {
+    // position an extra icon and fill to width of percentage
+    const rootVal = Math.floor(this.average);
+    const decimal = parseInt((this.average % 1).toFixed(2).split(".")[1]);
+
+    // track the closest full star to position the partial icon
+    this.ratingItems?.forEach((item) => {
+      if (this.average && !this.value && parseInt(item.dataset.value) === rootVal + 1) {
+        this.rootStar = item;
+        this.rootStar.dataset.partialparent = "true";
+      }
+    });
+
+    // only generate partial star if there is a root with a partial
+    if (this.rootStar) {
+      this.partialStar = document.createElement("calcite-icon") as HTMLCalciteIconElement;
+      this.partialStar.dataset.partial = "true";
+      this.partialStar.dataset.partialhidden = "false";
+      this.partialStar.dataset.rootvalue = (rootVal + 1).toString();
+      this.partialStar.scale = this.scale;
+      this.partialStar.theme = this.theme;
+      this.partialStar.icon = this.selectedIconType;
+      // edge legacy
+      this.partialStar.classList.add("calcite-rating-partial-star");
+
+      const ltrClip = `polygon(0 0, ${decimal}% 0, ${decimal}% 100%, 0% 100%)`;
+      const rtlClip = `polygon(${100 - decimal}% 0, 100% 0%, 100% 100%, ${100 - decimal}% 100%)`;
+
+      /*
+      // @ts-ignore-error
+      // this.partialStar.style.webkitClipPath = this.dir !== "rtl" ? ltrClip : rtlClip;
```

Comment:
I think it would probably be ok to do this:

```ts
// set webkit vendor specific prefix
(this.partialStar.style as any).webkitClipPath = this.dir !== "rtl" ? ltrClip : rtlClip;
```

Reply:
Updated to just use width ⬅️ ➡️